src/QMCHamiltonians/MomentumEstimator.cpp

```diff
@@ -172,115 +147,271 @@ void MomentumEstimator::setParticlePropertyList(PropertySetType& plist
   if (!hdf5_out)
   {
     copy(nofK.begin(),nofK.end(),plist.begin()+myIndex+offset);
-    copy(compQ.begin(),compQ.end(),plist.begin()+myIndex+nofK.size()+offset);
   }
 }
 
 bool MomentumEstimator::putSpecial(xmlNodePtr cur, ParticleSet& elns, bool rootNode)
 {
   OhmmsAttributeSet pAttrib;
   std::string hdf5_flag="yes";
+  kmax=0.0;
+  kmax0=0.0;
+  kmax1=0.0;
+  kmax2=0.0;
   pAttrib.add(hdf5_flag,"hdf5");
-  pAttrib.add(kgrid,"grid");
-  pAttrib.add(M,"samples");
+  //pAttrib.add(kgrid,"grid");
+  pAttrib.add(kmax,"kmax");
+  pAttrib.add(kmax0,"kmax0");
+  pAttrib.add(kmax1,"kmax1");
+  pAttrib.add(kmax2,"kmax2");
+  pAttrib.add(M,"samples"); // default value is 40 (in the constructor)
   pAttrib.put(cur);
   hdf5_out = (hdf5_flag=="yes");
+  RealType min_Length=1.0e10;
+  PosType vec_length(0);
+  for (int i=0; i<OHMMS_DIM; i++)
+  {
+    PosType a(0);
+    for (int j=0; j<OHMMS_DIM; j++)
+      a[j]=elns.Lattice.G(j,i);
```

Comment:
G=inverse(R). Not understanding why you take the columns instead of rows.

Reply:
In CrystalLattice.h "///Reciprocal unit vectors. G(j,i) i=vector and j=x,y,z".

Reply:
G=inverse(transpose(R))

Reply:
@jtkrogel where in the code?
I found CrystalLattice.cpp:113
```
  G = inverse(R); //G = transpose(Inverse(R));
```
transpose is commented

Reply:
Theoretically transpose(G)R=2pi x identity_matrix. In QMCPACK the 2pi is omitted. Therefore, inverse(R)=transpose(G)=[a,b,c], where the column vectors a, b, and c are the reciprocal space vectors.

Reply:
@kylanpaa You are right. The definition requires a transpose.

Reply:
Then please use Gv to simplify your code.